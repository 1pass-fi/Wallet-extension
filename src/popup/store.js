import { composeWithDevTools } from 'redux-devtools-extension'
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'

import rootReducer from './reducers/index'

const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
)

export default store
