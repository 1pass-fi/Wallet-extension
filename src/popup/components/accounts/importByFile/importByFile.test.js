import '@babel/polyfill'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import store from 'popup/store'
import React from 'react'

import ImportByFile from '.'

import { BrowserRouter as Router } from 'react-router-dom'

describe('Test for ImportByFile component', () => {
  describe('Render without crashing', () => {
    it('renders correctly', () => {
      const { container } = render(<Router><Provider store={store}><ImportByFile /></Provider></Router>)
      expect(container).toMatchSnapshot()
    })
  })
})
