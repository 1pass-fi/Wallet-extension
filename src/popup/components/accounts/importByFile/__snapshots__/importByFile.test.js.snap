// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ImportByFile component Render without crashing renders correctly 1`] = `
<div>
  <div
    class="account-import-key"
  >
    <div
      class="card import-card"
    >
      <div
        class="title"
      >
        <svg
          class="title-icon"
        />
        <div
          class="title-text"
        >
          Upload a .JSON file
        </div>
      </div>
      <form>
        <div
          class="dropzone"
          tabindex="0"
        >
          <div
            class="decorator"
          >
            <input
              accept="application/json"
              autocomplete="off"
              data-testid="fileInput"
              multiple=""
              name="fileField"
              style="display: none;"
              tabindex="-1"
              type="file"
            />
            <div
              class="drag-active"
            >
              <svg
                class="export-icon"
              />
              <div
                class="import-description"
              >
                Drag & drop an existing .JSON wallet file here or click to browse your computer
              </div>
            </div>
          </div>
        </div>
        <div
          class="create-password"
        >
          <div
            class="fields"
          >
            <div
              class="input-field "
            >
              <div
                class="label"
              >
                New password
              </div>
              <div
                class="input-wrapper"
              >
                <input
                  class="input"
                  name="pwd"
                  placeholder="Make it unique (min. 8 characters)"
                  type="password"
                  value=""
                />
                <svg
                  class="toggle-display"
                />
              </div>
            </div>
            <div
              class="input-field "
            >
              <div
                class="label"
              >
                Confirm password
              </div>
              <div
                class="input-wrapper"
              >
                <input
                  class="input"
                  name="pwdConfirm"
                  placeholder=""
                  type="password"
                  value=""
                />
                <svg
                  class="toggle-display"
                />
              </div>
            </div>
          </div>
          <div
            class="term-service"
          >
            <div
              class="checkbox"
            >
              <input
                name="checkbox"
                type="checkbox"
                value="false"
              />
            </div>
            <label>
              I agree with the 
              <a
                href="#"
              >
                Terms of Service
              </a>
            </label>
          </div>
          <div
            class="button"
          >
            <button
              class="button-shared button-filled import-button"
            >
              Import Wallet
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
